lib/acme/client/resources/account.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+class Acme::Client::Resources::Account
+  attr_reader :url, :status, :contact, :term_of_service, :orders_url
+
+  def self.arguments_from_response(response)
+    attributes = response.body
+    status = attributes.fetch('status')
+    term_of_service = attributes['termsOfServiceAgreed']
+    contact = attributes.fetch('contact', [])
+
+    { term_of_service: term_of_service, status: status, contact: contact }
+  end
+
+  def initialize(client, **arguments)
+    @client = client
+    assign_attributes(arguments)
+  end
+
+  def kid
+    url
+  end
+
+  def update(contact: nil, terms_of_service_agreed: nil)
+    assign_attributes **@client.account_update(
+      contact: contact, terms_of_service_agreed: term_of_service
+    ).to_h
+    true
+  end
+
+  def deactivate
+    assign_attributes **@client.account_deactivate.to_h
```

Comment:
If I'm reading this correctly,

1. `@client.account_deactivate` calls `arguments_from_response`
2. The returned hash is used to initialize a `Acme::Client::Resources::Account`
3. We then call `to_h` on that instance to merge those attributes into this instance

This sounds to me like the responsibilities of the `Account` class are spread all over the place and `@client` is doing too much.

Reply:
I'm trying to find a way to allow:

```ruby
client.account_deactivate(url: account_url)
```

But also allow:

```ruby
account = client.account(url: account_url)
account.deactivate
```

But without repeating myself too much. I'm not too worried about the `Client` class getting big, as long as it only do http stuff.

I think that exacting the argument from the response `arguments_from_response` is the flaw in this design, that's what make `account > client > account > client > account`. Do that make sense to you?

Reply:
I've refactor this [here](https://github.com/unixcharles/acme-client/pull/134/files#diff-e843384610a1a0cb8c21582bbed52188R205)